src/main/java/core/basesyntax/dao/ma/MentorDaoImpl.java

```diff
@@ -11,6 +16,15 @@ public MentorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public List<Mentor> findByAgeGreaterThan(int age) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Mentor> criteriaQuery = criteriaBuilder.createQuery(Mentor.class);
+            Root<Mentor> root = criteriaQuery.from(Mentor.class);
+            Predicate predicate = criteriaBuilder.gt(root.get("age"), age);
+            criteriaQuery.select(root).where(predicate);
+            return session.createQuery(criteriaQuery).getResultList();
```

Comment:
We can use here just HQL without CriteriaQuery

Reply:
thanks for advice!